### PR TITLE
Unit tests for models/pull.go

### PR DESCRIPTION
**Pull request by ethantkoenig**
~~The main problem is that I'm not sure the best way to "mock out" git repositories in the filesystem to match the database fixtures.~~

~~Should we add a script to create these repos? This seems less-than-ideal, since the script would be really tedious, and it would have to be kept in-sync with the test fixtures. Does anyone have a better suggestion?~~

I've removed all tests that depend on git, and replaced them with TODOs.

### CODE REVIEW

**Comment by andreynering** [comment]
Failing for me and on Drone.

I think mocking Git is not trivial and should be discussed in a proposal.

Can you for now don't keep tests that touch Git?

**Comment by ethantkoenig** [comment]
Sure

File: models/pull.go [hunk 1]
```diff
@@ -624,7 +631,7 @@ func GetPullRequestByIndex(repoID int64, index int64) (*PullRequest, error) {
 	if err != nil {
 		return nil, err
 	} else if !has {
-		return nil, ErrPullRequestNotExist{0, repoID, index, 0, "", ""}
+		return nil, ErrPullRequestNotExist{0, 0, 0, repoID, "", ""}
```
Comment (bkcsoft) [inline models/pull.go:649]: Why was this changed? :)

**Comment by bkcsoft** [comment]
Maybe we shouldn't mock git operations? maybe provide "well known" repos for testing?

File: models/pull.go (hunk 1 above)
Comment (ethantkoenig) [reply models/pull.go:649]: The arguments were provided in the wrong order. Sorry, maybe this should have been in a separate PR.

**Comment by ethantkoenig** [comment]
@bkcsoft @andreynering For now, I've added TODOs for the tests that require git.

**Comment by lunny** [comment]
Well known repo ---> Gitea

File: models/pull.go [hunk 2]
```diff
@@ -94,9 +95,11 @@ func (pr *PullRequest) loadAttributes(e Engine) (err error) {
 	if pr.HasMerged && pr.Merger == nil {
 		pr.Merger, err = getUserByID(e, pr.MergerID)
 		if IsErrUserNotExist(err) {
+			fmt.Printf("User doesn't exist! (mergerId=%d)\n", pr.MergerID)
```
Comment (lunny) [inline models/pull.go]: Unremoved trace code?

File: models/pull.go [hunk 3]
```diff
@@ -94,9 +95,11 @@ func (pr *PullRequest) loadAttributes(e Engine) (err error) {
...
 			pr.MergerID = -1
 			pr.Merger = NewGhostUser()
 		} else if err != nil {
+			fmt.Printf("loadAttirbutes(mergerId=%d): Error%v\n", pr.MergerID, err)
```
Comment (lunny) [inline models/pull.go]: ^

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by ethantkoenig** [comment]
@lunny Trace code removed, sorry about that

File: models/pull.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ import (
 	api "code.gitea.io/sdk/gitea"
 	"github.com/Unknwon/com"
 	"github.com/go-xorm/xorm"
+	"code.gitea.io/gitea/modules/base"
 )
```
Comment (andreynering) [inline models/pull.go:25]: Import grouping:

```go
 import (
 	"fmt"
 	"os"
 	"path"
 	"path/filepath"
 	"strconv"
 	"strings"
 	"time"
 
 	"code.gitea.io/git"
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/process"
 	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/sync"
  	api "code.gitea.io/sdk/gitea"
 	"code.gitea.io/gitea/modules/base"

  	"github.com/Unknwon/com"
  	"github.com/go-xorm/xorm"
  )
```

File: models/pull_test.go [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+	"time"
+)
```
Comment (andreynering) [inline models/pull_test.go]:
```go
import (
 	"testing"
 	"time"

 	"github.com/stretchr/testify/assert"
 )
```

**Comment by andreynering** [comment]
Two little things. Otherwise LGTM

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by strk** [comment]
trusted LGTM

**Pull request merged**